### PR TITLE
lib: lwm2m_carrier: update to version 0.8.0

### PR DESCRIPTION
**Pull request by lemrey**
Update library version to 0.8.0.

### CODE REVIEW

File: lib/bin/lwm2m_carrier/include/lwm2m_os.h [hunk 1]
```diff
@@ -21,16 +22,79 @@
...
+
+/**
+ * @brief AT Command handler.
+ */
+typedef void (*lwm2m_os_at_cmd_handler_t)(void *ctx, char *response);
+
+/**
+ * @brief Download client event IDs.
+ */
+enum lwm2m_os_download_evt_id {
```
Comment (rlubos) [inline lib/bin/lwm2m_carrier/include/lwm2m_os.h]: I'll just link my comment from previous PR here, for consideration: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1313#discussion_r337046714

File: lib/bin/lwm2m_carrier/include/lwm2m_os.h [hunk 2]
```diff
@@ -115,5 +179,137 @@ const char *lwm2m_os_log_strdup(const char *str);
...
+
+/**
+ * @brief Translate the error number.
+ */
+int lwm2m_os_errno(void);
+
 /**@} */
-#endif /* LWM2M_OS_H__ */
+
+#endif // LWM2M_OS_H__
```
Comment (rlubos) [inline lib/bin/lwm2m_carrier/include/lwm2m_os.h]: We should stick to C-style comments in NCS.

**Comment by lemrey** [comment]
Updated and rebased.
Removed commit adding 0.7.0 and reworked changelog accordingly.

**Comment by SebastianBoe** [comment]
This was merged without checking the DNM label by accident.

**Pull request merged**